src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,40 +1,92 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
+import { Todo } from '../../types/Todo';
+import { User } from '../../types/User';
 import { Loader } from '../Loader';
+import { getUser } from '../../api';
+
+type Props = {
+  todo: Todo;
+  onClose: () => void;
+};
+
+export const TodoModal: React.FC<Props> = ({ todo, onClose }) => {
+  const [user, setUser] = useState<User | null>(null);
+  const [loading, setLoading] = useState<boolean>(false);
+  const [error, setError] = useState<string | null>(null);
+
+  useEffect(() => {
+    const fetchUser = async () => {
+      setLoading(true);
+      setError(null);
+      try {
+        const fetchedUser = await getUser(todo.userId);
+
+        setUser(fetchedUser);
+      } catch (err) {
+        setError('Не вдалося завантажити користувача');
+      } finally {
+        setLoading(false);
+      }
+    };
+
+    fetchUser();
+  }, [todo.userId]);
 
-export const TodoModal: React.FC = () => {
   return (
     <div className="modal is-active" data-cy="modal">
-      <div className="modal-background" />
+      <div className="modal-background" onClick={onClose} />
```

Comment:
The naming convention for your event handler function could be improved. It's a good practice to name event handlers starting with 'handle' and ending with the event type. In this case, 'handleClose' would be a more appropriate name than 'onClose'. You can read more about this [here](https://medium.com/javascript-in-plain-english/handy-naming-conventions-for-event-handler-functions-props-in-react-fc1cbb791364).